src/library/dlgtagfetcher.cpp

```diff
@@ -80,11 +80,21 @@ void DlgTagFetcher::init() {
     }
     connect(btnApply, &QPushButton::clicked, this, &DlgTagFetcher::apply);
     connect(btnQuit, &QPushButton::clicked, this, &DlgTagFetcher::quit);
+    connect(btnRetry, &QPushButton::clicked, this, &DlgTagFetcher::retry);
     connect(results, &QTreeWidget::currentItemChanged, this, &DlgTagFetcher::resultSelected);
 
     connect(&m_tagFetcher, &TagFetcher::resultAvailable, this, &DlgTagFetcher::fetchTagFinished);
     connect(&m_tagFetcher, &TagFetcher::fetchProgress, this, &DlgTagFetcher::fetchTagProgress);
+    connect(&m_tagFetcher,
+            &TagFetcher::currentRecordingFetched,
```

Comment:
This signal can be extended to directly submit the result. This way we can populate the box step by step.